applications/serial_lte_modem/src/lwm2m_carrier/slm_at_carrier.c

```diff
@@ -148,18 +148,30 @@ static void reconnect_wk(struct k_work *work)
 int lwm2m_carrier_event_handler(const lwm2m_carrier_event_t *event)
 {
 	int err = 0;
-	static bool fota_started;
 
 	switch (event->type) {
+#if defined(CONFIG_SLM_CARRIER_AUTO_CONTROL)
 	case LWM2M_CARRIER_EVENT_LTE_LINK_UP:
 		LOG_DBG("LWM2M_CARRIER_EVENT_LTE_LINK_UP");
-		if (fota_started) {
-			fota_started = false;
-			k_work_reschedule(&reconnect_work, K_MSEC(100));
-		} else {
-			/* AT+CFUN=1 to be issued. */
-			err = -1;
-		}
+		k_work_reschedule(&reconnect_work, SLM_UART_RESPONSE_DELAY);
+		break;
+	case LWM2M_CARRIER_EVENT_LTE_LINK_DOWN:
+		LOG_DBG("LWM2M_CARRIER_EVENT_LTE_LINK_DOWN");
+		slm_util_at_printf("AT+CFUN=4");
```

Comment:
```suggestion
		err = slm_util_at_printf("AT+CFUN=4");
```
or maybe even (and same for the other occurences)
```suggestion
		err = slm_util_at_printf("AT+CFUN=4");
		if (err) {
			err = -1;
		}
```